### ISSUE TITLE
Collapse Block trick on Redstone Ore voids block when falling on non-full block

### ISSUE THREAD

**Issue by pjfranke** [Vazkii/Psi#171]
The collapse block trick makes blocks behave like sand/gravel, and free fall.  If the trick is used on Redstone Ore, with a solid block Y-2 or Y-3 of the target block, it will fall and land, and sparkle as redstone ore does when interacted with.

If a non-solid block or item (lever, torch, button, slab, etc) is Y-3 of the target block, it will fall, but instead of popping to an item as all other blocks normally do, the redstone ore is voided, as sparkling redstone ore does not have a valid item to drop.

**Comment by yrsegal** [comment]
Isn't this a vanilla bug?

**Comment by pjfranke** [comment]
Eeeh....sort of, but not really. It's true, that the sparkling Redstone ore not "popping" to an item entity,  is due to vanilla not having an item drop set for it . But in vanilla, no blocks would ever be affected by gravity, except for sand and gravel.

Being that Redstone ore is a special case when dealt with by the mod function (possibly the only one), it needs special treatment.

**Comment by pjfranke** [comment]
I recently realized that the "Collapse Block" trick is basically doing the same function the "Mana Lens Weight" does, which does behave correctly on Redstone Ore.  Well, after testing, I learned something.  Specifically, on "redstone_ore" ID 73, they both work fine.  Neither one actually works on "lit_redstone_ore" ID 74.  So it's necessary to wait for a Block Tick, and the redstone ore to un-light, before dropping it with either a Collapse Block spell, or a Mana Spreader/Blaster with the Weight lens attached.

Or, for maximum user friendly-ness, when either a Weight mana-burst or a Collapse Block Trick affect a Lit Restone ore block, swap it for a Redstone Ore, THEN drop the block.

**Comment by howtonotwin** [comment]
What about lit Redstone Lamps?

**Comment by Vazkii** [comment]
Does this also apply to them?

**Comment by yrsegal** [comment]
Yes, since they don't have an item form. Don't think there are any others like this in vanilla, although I may be wrong.

**Comment by howtonotwin** [comment]
There are furnaces too, but as long as TEs are not allowed it's fine.

**Comment by yrsegal** [comment]
This is how 1.9.4 botania does it, if that matters https://github.com/williewillus/Botania/blob/MC19/src/main/java/vazkii/botania/common/item/lens/LensWeight.java

**Comment by Vazkii** [comment]
The collapsing itself is not what triggers the redstone block changing, it's when you right click on it.
